sources/niLang/src/DataTablePath.cpp

```diff
@@ -321,20 +397,31 @@ cString __stdcall cDataTablePath::GetRootPathToDataTable(iDataTable* apDT, tU32
     niCheck(ret != _A("@"),AZEROSTR);
   }
 
-  if  (apDT->GetParent()) {
-    while (apDT->GetParent()) {
-      cString cur;
-      cur = _A("/");
-      cur += apDT->GetName();
-      cur += ret;
-      ret = cur;
-      apDT = apDT->GetParent();
+  while (apDT->GetParent()) {
+    const cString childName = apDT->GetName();
+    tU32 childIndex = 0;
+    niLoop(i, apDT->GetParent()->GetNumChildren()) {
+      auto child = apDT->GetParent()->GetChildFromIndex(i);
+      if (child == apDT) {
+        break;
+      }
+      if (childName.Eq(child->GetName())) {
+        ++childIndex;
+      }
     }
-  }
-  else {
-    ret = cString(_A("/")) + ret;
-  }
+    cString displayedName = childName;
```

Comment:
A way to make this a bit more performant is to think of cString like a StringBuilder in Java. That is put it outside of the block and .clear() before using it. clear() doesnt deallocate memory so you'll avoid a bunch of allocation this way.

I think you can just concat in ret directly? ret.clear() and then what you're doing with displayedName should be equivalent if I didnt missread.